rust/datafusion/src/scalar.rs

```diff
@@ -72,12 +77,14 @@ pub enum ScalarValue {
     Date32(Option<i32>),
     /// Date stored as a signed 64bit int
     Date64(Option<i64>),
+    /// Timestamp Second
+    TimestampSecond(Option<i64>),
```

Comment:
I added support for `TimestampSecond` here and renamed `ScalarValue::TimeMillisecond` --> `ScalarValue::TimestampMillisecond` so that the `ScalarValue` enum type matches up with the `DataType` enum type name

Reply:
👍  Great to be consistent.
Not related to this PR necessarily but I'm going to work on support for converting date strings to timestamps of different resolutions.  Right now only Nanos are supported, ideally we'd have `to_timestamp(...)` that allows choosing nanos, micros, millis, or seconds under the hood.